wazuh/server/ssh-key-logging/Extending_The_SSH_Logger.MD

```diff
@@ -0,0 +1,146 @@
+
+```
+<ruleset>
+...
+<decoder_exclude>ruleset/decoders/0310-ssh_decoders.xml</decoder_exclude>
+</ruleset>
+```
+
+* Now the edits and customisations to the new customised version of the ssh decoder is made, at this point its worth identifying a log entry that you would like to be captured, the sample used for extending ssh logins was extracted from the auth.log file and is detailed below:
+
+```
+Nov 19 21:33:32 xxxxxx sshd[1914646]: Accepted publickey for xxxxxx from xxx.xx.xxx.xxx port xxxxx ssh2: RSA SHA256:XXXXXXXXX
+```
+
+The following changes are being made to the new custom/copied decoder file (/var/ossec/etc/decoders/0310-ssh_decoders.xml)
+The decoder works by using a regex to parse logfile entries for matches, and then an offset to capture data. Captured fields are shown by the (\S+) strings in the decoder, with the order element of the xml assigning those strings to variables
+
+The default decoder for ssh successful ssh login is detailed below, and as you can see from the example log line, the first string after "for" is allocated to the name the second string element from (\S+) is recorded into the srcip field and the port (\S+) is recorded into the srcport variable.
+
+```
+<decoder name="sshd-success">
+  <parent>sshd</parent>
+  <prematch>^Accepted</prematch>
+  <regex offset="after_prematch">^ \S+ for (\S+) from (\S+) port (\S+)</regex>
+  <order>user, srcip, srcport</order>
+  <fts>name, user, location</fts>
+</decoder>
+```
+
+In order to update this to capture the SSH key checksum you can see the "regex offset" element gets updated to look for the RSA SHA256 string before assigning that string to a new custom field called "keysum". This field will become available in Wazuh without any further configuration.
+
+```
+<decoder name="sshd-success">
+  <parent>sshd</parent>
+  <prematch>^Accepted</prematch>
+  <regex offset="after_prematch">^ \S+ for (\S+) from (\S+) port (\S+) ssh2: RSA SHA256:(\S+)</regex>
+  <order>user, srcip, srcport, keysum</order>
+  <fts>name, user, location</fts>
+</decoder>
+```
+
+* Once these changes have been made, it is necessary to restart the Wazuh manager so the changes take effect:
+
+```
+systemctl restart wazuh-manager
+```
+
+If there are any issues with restarting the server, which can be caused by syntax errors in either the config or decoder files, these can be identified by running a status on the Wazuh manager service.
+
+```
+systemctl status wazuh-manager.service
+```
+
+## Testing The New Decoder/Changes
+
+Wazuh provide a useful tool for checking whether decoder and rule changes are being parsed properly, this can be used prior to any changes to see what rules etc are being parsed, and what data is being captured. Below is the output of the test of the ssh decoder test detailed above:
```

Comment:
```suggestion
Wazuh provides useful tools for checking whether decoder and rule changes are being parsed properly. This can be used prior to any changes to see what rules (etc) are being parsed, and what data is being captured. Below is the output of the ssh decoder test detailed above:
```